openml/extensions/extension_interface.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional, Tuple
+
+from openml import OpenMLTask, OpenMLFlow
+from openml.runs.trace import OpenMLTraceIteration, OpenMLRunTrace
+
+
+class Extension(ABC):
```

Comment:
I had left some comments here before but they fell off when I edited the file.. Anyhow, this way to define an abstract base class is [preferred as of Python 3.4](https://docs.python.org/3/whatsnew/3.4.html#abc). See also [this](https://docs.python.org/3/library/abc.html#abc.ABC).